speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp

```diff
@@ -0,0 +1,87 @@
+#include "Recorder.h"
+#include <iomanip>
+#include <sstream>
+#include <iostream>
+#include <tgf.h>
+
+///
+/// @brief Constructor of Recording, 
+/// creates a file with the current date and time,
+/// opens file in binary and appending mode
+/// records the start time of creation,
+/// initializes the previous input with 2's since user input is between -1 an 1.
+/// also creates a folder for the files
+/// @author Sjoerd Schilder
+///
+Recorder::Recorder()
+{
+	// create directory
+	std::string dirName = "..\\test_data\\user_recordings";
+	if(!GfDirExists(dirName.c_str()))
+		GfDirCreate(dirName.c_str());
+	// get current date and time
+	std::time_t t = std::time(nullptr);
+	std::tm tm = *std::localtime(&t);
+	std::stringstream buffer;
+	buffer << std::put_time(&tm, "%Y%m%d-%H%M%S");
+	// create file
+	std::string fileName = dirName + "\\Record" + buffer.str() + ".txt";
+	m_recordingFile.open(fileName, std::ios::binary | std::ios::app);
+	// start clock
+	m_startTime = clock();
```

Comment:
Leave some blank space before a singleline-comment and not every line needs a comment
```suggestion
	// create directory if it doesn't exist
	std::string dirName = "..\\test_data\\user_recordings";
	if(!GfDirExists(dirName.c_str())) 
	{
		GfDirCreate(dirName.c_str());
	}
	
	// create and open new recording file with current timestamp
	std::time_t t = std::time(nullptr);
	std::tm tm = *std::localtime(&t);
	std::stringstream buffer;
	buffer << std::put_time(&tm, "%Y%m%d-%H%M%S");
	std::string fileName = dirName + "\\Record" + buffer.str() + ".txt";
	m_recordingFile.open(fileName, std::ios::binary | std::ios::app);

	m_startTime = clock();
```